protocol/RESP3.md

```diff
@@ -270,14 +270,22 @@ However the client library should return a floating point number in one
 case and an integer in the other case, if the programming language in which
 the client is implemented has a clear distinction between the two types.
 
-In addition the double reply may return positive or negative infinity,
-and positive or negative NaN as the following four stings:
+In addition, the double reply may return positive or negative infinity
+as the following two stings:
 
     ",inf\r\n"
     ",-inf\r\n"
+
+In addition, before Redis 7.2, the double reply may return positive or negative NaN
+as the following two stings:
+
     ",nan\r\n"
     ",-nan\r\n"
 
+Starting from Redis 7.2, any form of NaN will be normalized to a single nan string:
+
+    ",nan\r\n"
+
```

Comment:
I'm not sure the history of Redis needs to be part of the RESP specification. Perhaps it can mentioned in a note or in parentheses?

Btw, before Redis 7.2 I think the nan could be any representation returned by libc, for example "-NAN", "nan(payload)", etc.

Reply:
yes, i was hesitant about the history too. the reason i add it is because oran says:

> also document the other types (specifically the negative one) in the protocol as a backwards compatibility thing.

> Btw, before Redis 7.2 I think the nan could be any representation returned by libc, for example "-NAN", "nan(payload)", etc.

that is right, do you have any good words for me to use? i am worry about my wording...

Reply:
I think we should mention in the Version history of the RESP3 protocol (top of this page) this change and mark a new version of the document. Until now, the specification said "nan" or "-nan", but from now on only "nan".

Idea for version history:

> * 1.4, xx Dec 2022, Normalize NaN to a single representation "nan" and forbid "-nan".

Then, in the details about number, we first specify the protocol and afterwards mention old versions of the protocol and versions of Redis.

Idea:

> In addition the double reply may return positive or negative infinity, and NaN as the following three stings:
>
>     ",inf\r\n"
>     ",-inf\r\n"
>     ",nan\r\n"
>
> In earlier versions of this specification, `,-nan\r\n` "negative NaN" was also valid. Note that Redis prior to version 7.2 may return any representation of NaN produced by libc, such as "-nan", "NAN" or "nan(char-sequence)".